src/presign/participant.rs

```diff
@@ -399,24 +570,39 @@ impl PresignParticipant {
         message: &Message,
         main_storage: &Storage,
     ) -> Result<(Option<PresignRecord>, Vec<Message>)> {
+        // If we have not yet started round three, stash the message for later
+        let r3_started = self
+            .storage
+            .retrieve(
+                StorableType::PresignRoundThreePrivate,
+                message.id(),
+                self.id,
+            )
+            .is_ok();
```

Comment:
thought: Is it efficient to retrieve stuff from storage? It looks like you use this value in `do_presign_finish` below -- when we get into optimizing, we might save the value here and pass it to that function instead of accessing storage twice. Ditto with accessing the round 3 public values from other participants.

To be clear: I don't think we have to change anything in this PR.

Reply:
Hmm... while I don't think storage retrieval is very expensive relatively speaking right now, it does involve deserializing and in the long term, would probably involve accessing hard drive storage, so it is probably worth considering easy optimizations like this that don't majorly complicate the code. I'll keep this in mind, thanks!